sbin/upload-salt-reports

```diff
@@ -34,17 +34,26 @@ def salt_config():
 
 def get_job(job_id):
     result = run('jobs.lookup_jid', [job_id])
-
     # If any minion's results are strings, they're exceptions
     # and should be wrapped in a list like other errors
+
     for minion, value in result.items():
-        if type(value) == str:
-            result[minion] = [value]
-        else:
-            for key, entry in value.items():
-                if key.startswith('module_') and entry['__id__'] == 'state.highstate':
-                    result[minion] = entry['changes']['ret']
-                    break
+        try:
+            if isinstance(value,str):
+                result[minion] = [value]
+            elif isinstance(value,list):
+                result[minion] = value
+            else:
+                for key, entry in value.items():
+                    try:
+                        if key.startswith('module_') and entry['__id__'] == 'state.highstate':
+                            result[minion] = entry['changes']['ret']
+                            break
+                    except:
+                        print("Failed to process key %s for minion %s" % (key, minion))
+        except:
+            print("Failed to process data for minion %s" % minion)
+            print("The value is: %s of type: %s" % (value, type))
```

Comment:
Is this something you want or did you forget to remove this "kind of debug" information? As said previously, it would be good to print the exception. 

Is it useful to add a finally: block which return some fallback data which hopefully will be ignored by foreman?